### PR TITLE
PART 2: Add proper command line parsing for all scripts - Based on #1970

### PR DESCRIPTION
**Pull request by okurz**
Based on:
* https://github.com/os-autoinst/os-autoinst/pull/1970

This also ensures that scripts are included in coverage analysis

In https://github.com/os-autoinst/os-autoinst/pull/1970 I realized that we don't have any coverage for most scripts in the top-level folder.

### CODE REVIEW

**Review by baierjan — APPROVED** [review]

**Review by Martchus — COMMENTED** [review]
Have you tested the openvswitch script in action? I'm just asking because I somehow don't trust this "magic" D-Bus invocations. (The specification of arguments seems rather explicit so I *guess* it should work.)

File: os-autoinst-openvswitch [hunk 1]
```diff
@@ -20,12 +20,22 @@ package OVS;
...
 use autodie ':all';
+use Getopt::Long;
 use Net::DBus::Exporter 'org.opensuse.os_autoinst.switch';
 require IPC::System::Simple;
 use IPC::Open3;
 use Symbol 'gensym';
 use Time::Seconds;
 
+sub usage ($r) {
+    eval { require Pod::Usage; Pod::Usage::pod2usage($r) };
```
[the remark this answers is not in the document]
Comment (okurz) [reply os-autoinst-openvswitch:31]: Yes, I left it out on purpose

**Comment by okurz** [comment]
> Have you tested the openvswitch script in action? I'm just asking because I somehow don't trust this "magic" D-Bus invocations. (The specification of arguments seems rather explicit so I _guess_ it should work.)

I have kinda done that by just calling `os-autoinst-openvswitch` locally without argument. That fails the same way as in before with

```
org.freedesktop.DBus.Error.AccessDenied: Connection ":1.3390" is not allowed to own the service "org.opensuse.os_autoinst.switch" due to security policies in the configuration file
```

so it certainly does the same "magic" initial D-Bus invocations ;)

File: check_qemu_oom [hunk 2]
```diff
@@ -13,11 +13,20 @@ check_qemu_oom qemu_pid
...
+use Getopt::Long;
 
-my $qemu_pid = $ARGV[0];
+Getopt::Long::Configure("no_ignore_case");
+
+my %options;
 
-eval { use Pod::Usage; pod2usage(1); } unless $qemu_pid;
+sub usage ($r) {
+    eval { require Pod::Usage; Pod::Usage::pod2usage($r); };
```
Comment (baierjan) [inline check_qemu_oom:24]: Nitpick: an extra `;`

Comment (okurz) [reply check_qemu_oom:24]: Yep, you got me

File: check_qemu_oom [hunk 3]
```diff
@@ -13,11 +13,20 @@ check_qemu_oom qemu_pid
...
+    die "cannot display help, install perl(Pod::Usage)\n" if $@;
+}
 
-exit 0 if (index(qx{dmesg}, "Out of memory: Killed process $qemu_pid") != -1);
-exit 1;
+GetOptions(\%options, 'help|h|?') or usage(1);
+usage(0) if $options{help};
+usage(1) unless @ARGV;
+my $qemu_pid = $ARGV[0];
+exit(index(qx{dmesg}, "Out of memory: Killed process $qemu_pid") != -1) ? 0 : 1;
```
[the remark this answers is not in the document]
Comment (okurz) [reply check_qemu_oom:32]: Sorry, I don't understand what you mean. Which process should have terminated in before?

[the remark this answers is not in the document]
Comment (Martchus) [reply check_qemu_oom:32]: I've created https://github.com/os-autoinst/os-autoinst/pull/1975 so the problem can be considered out of scope here.

File: t/44-scripts.t [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+my %types = @types;
+for my $key (keys %types) {
+    delete $types{$key} unless $allowed_types{$types{$key}};
+}
+
+for my $script (sort keys %types) {
+    my $out = qx{timeout 3 $Bin/../$script --help 2>&1};
+    my $rc = $?;
+    is($rc, 0, "Calling '$script --help' returns exit code 0") or diag "Output: $out";
```
[the remark this answers is not in the document]
Comment (okurz) [reply t/44-scripts.t:29]: Hm, likely my latest commit caused this then

[the remark this answers is not in the document]
Comment (okurz) [reply t/44-scripts.t:29]: I have removed the latest commit for now.

**Pull request merged**